zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/RequestMapper.java

```diff
@@ -125,6 +129,24 @@ public static Either<ProblemDetail, ErrorJobRequest> toJobErrorRequest(
                 getMapOrEmpty(errorRequest, JobErrorRequest::getVariables)));
   }
 
+  public static Either<ProblemDetail, CorrelateMessageRequest> toMessageCorrelationRequest(
+      final MessageCorrelationRequest correlationRequest, final boolean multiTenancyEnabled) {
+    final var validationErrorResponse = validateMessageCorrelationRequest(correlationRequest);
```

Comment:
🔧 How about we validate the provided tenant id here or in the `validateMessageCorrelationRequest`? That way, any issues due to invalid input data would consistently lead to a `ProblemDetail`. Currently, an exception is thrown from this method if the tenant id is not valid, if I'm not mistaken?

Reply:
🤔 It does, but interestingly enough it does return a ProblemDetail so it seems it gets caught somewhere. Maybe some Spring magic?

But I agree we shouldn't have magic here. I wanted to put it in the `MessageCorrelateValidator` at first but ran into the problem that apart from validating, this also makes some decision on what the `tenantId` would be. So that makes it troublesome since we only return a `ProblemDetail` there. I'll see if I can think of something else.

Reply:
> 🤔 It does, but interestingly enough it does return a ProblemDetail so it seems it gets caught somewhere. Maybe some Spring magic?

Yeah, we have a global exception handler that catches exceptions as a last resort and returns HTTP 400 problem details.

> But I agree we shouldn't have magic here. I wanted to put it in the MessageCorrelateValidator at first but ran into the problem that apart from validating, this also makes some decision on what the tenantId would be. So that makes it troublesome since we only return a ProblemDetail there. I'll see if I can think of something else.

I see, yeah. The tenant validator could return an Either with either the tenant id to use or validation errors, right?